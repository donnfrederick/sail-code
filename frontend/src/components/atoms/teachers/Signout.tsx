import * as React from 'react'
import styled from 'styled-components'
import isHuawei from 'utils/isHuawei'

interface Props {
  marginBottom?: number
  onClick(): void
}

export default (props: Props) => {
  const { marginBottom = 0, onClick } = props

  return (
    <Container marginBottom={marginBottom} onClick={onClick}>
      ログアウト
    </Container>
  )
}

const Container = styled<Props, any>('div')`
  width: 100%;
  height: 135px;
  margin-bottom: ${props => props.marginBottom}px;
  background-color: #ffffff;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 0 5px 0 rgba(5, 68, 102, 0.1)'};
  font-size: 32px;
  font-weight: 500;
  line-height: 135px;
  letter-spacing: 0px;
  text-align: center;
  color: #138efd;
`
