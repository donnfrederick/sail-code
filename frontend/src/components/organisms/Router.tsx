import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
