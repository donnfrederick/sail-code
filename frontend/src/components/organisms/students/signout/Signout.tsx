import SignoutModal from 'components/organisms/students/modal_contents/signout'
import * as React from 'react'
import styled from 'styled-components'
import isHuawei from 'utils/isHuawei'

interface Props {
  openModal(): void
  setModalContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const { openModal, setModalContents } = props

  return (
    <Signout
      onClick={() => {
        setModalContents(<SignoutModal />)
        openModal()
      }}
    >
      Sign Out
    </Signout>
  )
}

const Signout = styled.div`
  width: 100%;
  height: 100px;
  margin: 80px 0;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 0 6px 0 rgba(5, 68, 102, 0.1)'};
  background-color: #ffffff;
  font-size: 32px;
  font-weight: 500;
  line-height: 100px;
  letter-spacing: 0.5px;
  text-align: center;
  color: #138efd;
`
