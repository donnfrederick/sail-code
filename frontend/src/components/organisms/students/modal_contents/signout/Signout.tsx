import Button from 'components/atoms/students/Button'
import ModalTextContainer from 'components/molecules/students/ModalTextContainer'
import { store } from 'components/organisms/Router'
import * as React from 'react'
import { push } from 'react-router-redux'
import styled from 'styled-components'
import clearAuthToken from 'utils/clearAuthToken'

interface Props {
  authToken: string
  close(): void
  deleteSignout(authToken: string): Promise<void>
}

export default (props: Props) => {
  const { authToken, close, deleteSignout } = props

  return (
    <Container>
      <ModalTextContainer heading={'Are you sure you want\nto sign out?'} />
      <ButtonContainer>
        <Button
          type="blue"
          text="Cancel"
          width={244}
          height={88}
          onClick={() => close()}
        />
        <Button
          type="white"
          text="Sign Out"
          width={244}
          height={88}
          onClick={async () => {
            try {
              await deleteSignout(authToken)
              clearAuthToken()
              close()
              // localStorage.clear()
              store.dispatch(push('/'))
              window.location.reload()
            } catch (error) {
              // tslint:disable-next-line
              console.error(error)
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`
