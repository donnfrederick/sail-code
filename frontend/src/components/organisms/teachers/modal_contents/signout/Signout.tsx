import Button from 'components/atoms/teachers/Button'
import ModalTextContainer from 'components/molecules/teachers/ModalTextContainer'
import { store } from 'components/organisms/Router'
import * as React from 'react'
import { push } from 'react-router-redux'
import styled from 'styled-components'
import clearAuthToken from 'utils/clearAuthToken'

interface Props {
  authToken: string
  close(): void
  deleteSignout(authToken: string): Promise<void>
}

export default (props: Props) => {
  const { authToken, close, deleteSignout } = props

  return (
    <Container>
      <ModalTextContainer heading={'本当にログアウトしますか？'} />
      <ButtonContainer>
        <Button
          type="blue"
          text="いいえ"
          width={272}
          height={112}
          onClick={() => close()}
        />
        <Button
          type="white"
          text="はい"
          width={272}
          height={112}
          onClick={async () => {
            try {
              await deleteSignout(authToken)
              clearAuthToken()
              close()
              store.dispatch(push('/'))
              window.location.reload()
            } catch (error) {
              // tslint:disable-next-line
              console.error(error)
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`
