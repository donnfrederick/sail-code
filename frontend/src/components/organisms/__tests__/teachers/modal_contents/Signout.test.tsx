import Router from 'components/organisms/Router'
import Signout from 'components/organisms/teachers/modal_contents/signout'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <Signout />', () => {
  const tree = renderer
    .create(
      <Router>
        <Signout />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
