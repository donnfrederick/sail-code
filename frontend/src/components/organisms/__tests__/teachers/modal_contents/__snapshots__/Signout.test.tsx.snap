// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Signout /> 1`] = `
<div
  className="sc-jTzLTM dMEaru"
>
  <div
    className="sc-VigVT kaYUeB"
  >
    <div
      className="sc-gZMcBi fxOrEW"
    >
      本当にログアウトしますか？
    </div>
  </div>
  <div
    className="sc-fjdhpX eIDOHw"
  >
    <button
      className="sc-bZQynM iiToMd"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <div
        className="sc-iwsKbI lgKvmV"
      >
        いいえ
      </div>
    </button>
    <button
      className="sc-EHOje dyvxUy"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <div
        className="sc-iwsKbI lgKvmV"
      >
        はい
      </div>
    </button>
  </div>
</div>
`;
